packages/dashboard-core-plugins/src/MarkdownPlugin.tsx

```diff
@@ -27,16 +27,27 @@ export function MarkdownPlugin(props: MarkdownPluginProps): JSX.Element | null {
     if (componentState != null) {
       ({ panelState = null } = componentState as MarkdownComponentState);
     }
-    if (
-      title == null ||
-      panelState == null ||
-      panelState.content == null ||
-      panelState.content.length === 0 ||
-      panelState.content === MarkdownUtils.DEFAULT_CONTENT
-    ) {
+    if (title == null || panelState == null || panelState.content == null) {
       // We don't want to save it if there's no content
       return null;
     }
+    if (panelState.content === '') {
+      // If the content is empty, display markdown panel with default content
+      const configNew = {
+        type: 'react-component' as const,
+        component: MarkdownPanel.COMPONENT,
+        props: {
+          id: configProps.id,
+          metaData: {},
+          panelState: {
+            content: MarkdownUtils.DEFAULT_CONTENT,
+          },
+        },
+        localDashboardId: configProps.id,
+        title: MarkdownUtils.DEFAULT_TITLE,
+      };
+      return hydrate(configNew);
+    }
```

Comment:
We should never be calling `hydrate` from within a dehydration function.

Dehydration is taking a panel and outputting a serializable object such that we can serialize the layout and save it; hydration is the opposite, where we take a serialized layout, and hydrate it into objects (such as a grid model).

When we're dehydrating the layout, we check if the returned dehydration is `null`, and if it is, we omit the component from the layout completely: https://github.com/deephaven/web-client-ui/blob/6cf124012a19b122b3a96f620886b970c12b6d29/packages/dashboard/src/layout/LayoutUtils.ts#L340

In this case though, we _do_ want to save the panel to the layout, we just don't want to save the actual content with it. I don't think there's a case where we would want to drop it completely from the layout (which returning `null` does).

Digging into it a bit ... I think you should actually remove this custom `dehydrateMarkdown` function completely, and just use the default hydrate/dehydrate functions (omitting the last two parameters from the call to `registerComponent` below). Then need to change `MarkdownPanel.getClosedMarkdowns` to filter out any panel that doesn't have a markdown set.

Actually found another bug too - if we return `null` from the dehydration function (as we're doing above), PanelManager throws an error:
```
Error while emitting event: TypeError: Cannot set properties of null (setting 'parentStackId')
    at PanelManager.addClosedPanel (PanelManager.ts:365:54)
    at PanelManager.handleClosed (PanelManager.ts:347:10)
    at EventHub.emit (EventEmitter.ts:69:22)
```
You can see this by just opening and closing a markdown panel before your changes. We should have a check there for if the dehydrated component is `null`, and if it is, skip pushing it onto the `closed` array.

Reply:
Thanks for the information on hydration was very helpful! Will make the changes and will take a look at the other bug you mentioned